src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java

```diff
@@ -51,7 +51,10 @@
     /**
      * The class loader to load external classes. Not initialized as this must
      * be set by my creator.
+     * @deprecated Checkstyle is not type aware tool and all class loading is potentially
+     *     unstable.
      */
+    @Deprecated
     private ClassLoader classLoader;
```

Comment:
can we start communication thread with xwiki ?
to ask them to be null aware ? or simply use classloader not from Check ? or disable their Check ? or ...

this will let us, remove functonality and keep only compilation compatibility. But there will be a lot of damage in world that we do not know, same as in xwiki.

Reply:
I created an issue in their tracker.
https://jira.xwiki.org/browse/XCOMMONS-1758

Reply:
If no disablement is done for xwiki , how cu is green now ?

Reply:
@romani See https://github.com/checkstyle/checkstyle/pull/7114#issuecomment-536144626
We didnt touch anything that prevents their code from compiling.

Reply:
xwiki merged their change. https://github.com/xwiki/xwiki-commons/commit/824156d82cfe9e0b64fbac88f404b887d00a6920

Reply:
Awesome!
And now I think we need to remove it from AbstractCheck.
Nobody monitoring Deprecated method usage in code, so we can damage others code right now.
We only keep it in Checker/TreeWalker only because we need this compatibility as we depend on maven plugin. All other should start migrating off classloader usage. As we ignore set of Classloader our change will damage behavior of some third party Checks ... it is ok ... they need to upgrade/disable/... .

Reply:
Done.